pkg/ee/applications/util.go

```diff
@@ -0,0 +1,115 @@
+/*
+                  Kubermatic Enterprise Read-Only License
+                         Version 1.0 ("KERO-1.0”)
+                     Copyright © 2024 Kubermatic GmbH
+
+   1.	You may only view, read and display for studying purposes the source
+      code of the software licensed under this license, and, to the extent
+      explicitly provided under this license, the binary code.
+   2.	Any use of the software which exceeds the foregoing right, including,
+      without limitation, its execution, compilation, copying, modification
+      and distribution, is expressly prohibited.
+   3.	THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package applications
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+
+	semverlib "github.com/Masterminds/semver/v3"
+	"gopkg.in/yaml.v3"
+
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	kubernetesutil "k8c.io/kubermatic/v2/pkg/provider/kubernetes"
+
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ErrNoClusterForTemplating = fmt.Errorf("no cluster found for template data")
+
+// TemplateData is the root context injected into each application values.
+type TemplateData struct {
+	Cluster ClusterData
+}
+
+// ClusterData contains data related to the user cluster
+// the application is rendered for.
+type ClusterData struct {
+	Name string
+	// HumanReadableName is the user-specified cluster name.
+	HumanReadableName string
+	// OwnerEmail is the owner's e-mail address.
+	OwnerEmail string
+	// ClusterAddress stores access and address information of a cluster.
+	Address kubermaticv1.ClusterAddress
+	// Version is the exact current cluster version.
+	Version string
+	// MajorMinorVersion is a shortcut for common testing on "Major.Minor" on the
+	// current cluster version.
+	MajorMinorVersion string
+}
```

Comment:
This seems inspired by our addons and I think we need to document it in a similar fashion, otherwise it's impossible for users to make use of the variables. Checkout `codegen/godoc/` and `hack/ci/update-docs.sh` and extend as necessary.